book/src/ui/state_interaction.md

```diff
@@ -0,0 +1,114 @@
+## State Interaction
+
+Let's declare our state, and call it `MenuState`: 
+
+```rust
+#[derive(Default)]
+pub sturct MenuState {
+	button: Option<Entity>,
+}
+```
+
+We give it a field named `button` which will hold an entity wrapped in
+an `Option<T>`. This simplifies things since we can now derive `Default`
+trait on it and we can make it as our initial state that the application
+will start off as.
+
+It will also serve to hold our ui entitiy.
+
+In our `on_start` method of this state we can create the button as shown in
+previous chapters, but here we will save the entitiy in our struct: 
+
+```rust
+fn on_start(&mut self, data: StateData<()>) {
+    let world = data.world; 
+
+    /* Create the transform */ 
+    let ui_transform = UiTransform { . . . };
+    
+    /* Create the text */
+    let ui_text = UiText { . . . };
+
+    /* Building the entity */
+   	let btn = world.create_entity()
+        .with(ui_transform)
+        .with(ui_text)
+		.with(Interactable)
+        .build();
+
+	/* Saving the button in our state struct */
+	self.button = Some(btn);
+}
+```
+
+All the input received will be handled in the [handle_event](https://docs.amethyst.rs/master/amethyst/trait.State.html#method.handle_event)
+method of our state: 
+
+```rust
+fn handle_event(
+	&mut self, 
+	_data: StateData<'_, GameData<'_, '_>>, 
+	event: StateEvent) -> SimpleTrans {
+	if let StateEvent::Ui(ui_event) = event {
+
+		let cond = ui_event.target == self.button.unwrap();
+
+		match ui_event.event_type {
+			UiEventType::Click if cond => { 
+				/* . . . */
+			},
+			_ => {
+				return SimpleTrans::None; 
+			},  
+		};
+	}
+
+	SimpleTrans::None
+}
+```
+We only care about the `UiEvent`s here, that's why there's no match statment.
```

Comment:
might be confusing to say there's no match statement (when there is one inside).
perhaps:

```md
We only care about the `UiEvent`s here, that's why we can use the `if-let` pattern.
```